CONTRIBUTING.md

```diff
@@ -1,17 +1,50 @@
-### Linee guida per lo sviluppo
-
- - Board per gestire i task (TODO)
- - I task devono essere semplici e descrittivi in modo che sia chiaro il lavoro da svolgere
- - **Non** eseguire il push di commit direttamente verso il branch main
- - Implementazioni monotematiche: creare un feature branch con nome possibilmente descrittivo (es: id dell’issue + brevissima descrizione, 123-feature-foo)
- - Implementazioni articolate: creare un integration branch (es: integration/feature-foo) da cui poi forkare ulteriori feature branches specifici (es: 123-feature-foo-bar)
- - Creare pull requests più semplici possibile per facilitare la review
- - Prima di mergiare verso main (o integration branch) di seguito `destinazione`:
+# Linee guida per lo sviluppo
+
+## Indice
+
+- [**Nomenclatura dei file**](#nomenclatura-dei-file)
+- [**URL**](#url)
+- [**Workflow di lavoro**](#workflow-di-lavoro)
+
+## Workflow di lavoro
+
+### Task
+
+- Board per gestire i task (TODO)
+- I task devono essere semplici e descrittivi in modo che sia chiaro il lavoro da svolgere
+
+### GIT
+
+- **Non** eseguire il push di commit direttamente verso il branch main
+- Implementazioni monotematiche: creare un feature branch con nome possibilmente descrittivo (es: id dell’issue + brevissima descrizione, 123-feature-foo)
+- Implementazioni articolate: creare un integration branch (es: integration/feature-foo) da cui poi forkare ulteriori feature branches specifici (es: 123-feature-foo-bar)
+- Creare pull requests più semplici possibile per facilitare la review
+- Prima di mergiare verso main (o integration branch) di seguito `destinazione`:
     - eseguire un pull del branch di destinazione `git checkout {destinazione} && git pull`, 
     - tornare al feature branch e ribasare `git checkout - && git rebase {destinazione}`
     - risolvere eventuali conflitti
```

Comment:
io qui oltre a risolvere i conflitti, direi anche di squashare tutti i commit in uno unico molto più descrittivo, avere nel main un sacco di commi del tipo "wip - bla bla" non è bellissimo, soprattutto nel caso in cui sia necessario fare il revert di qualcosa

Reply:
d'accordissimo

Reply:
Non sono molto d'accordo in quanto, IMHO, è più facile "navigare" indietro nella storia, qualora fossero state introdotte delle regressioni, alla ricerca delle stesse...
Se il commit è *squashato* addio ricerca!  
Oltre al fatto che, di nuovo, si perde la visione di cosa è stato introdotto, come e quando (e perché, molto spesso)...

In ogni caso, usando le PR, salta subito all'occhio su quale commit fare il `revert`, qualora si volesse fare il revert dell'intera PR... Anche se ci fossero centinaia di commit!
Sarebbe semplicemente il più vecchio tra i due parent del commit di merge. 😉

Ma dev'essere una visione distopica mia dell'avere l'**assoluto** controllo sui commit e sulla storia del repository...
> _Mi rendo conto solo ora di quanto ho bacchettato i miei colleghi e di come li ho educati bene all'uso di GIT..._ 🤔🤣

---

Scherzi a parte, mi adeguo... No problem...  
Correggo.